--- a/drivers/net/wireless/realtek/rtl8189fs/os_dep/linux/ioctl_cfg80211.c
+++ b/drivers/net/wireless/realtek/rtl8189fs/os_dep/linux/ioctl_cfg80211.c
@@ -5672,3 +5672,3 @@
-static int      cfg80211_rtw_dump_station(struct wiphy *wiphy, struct net_device *ndev,
+static int      cfg80211_rtw_dump_station(struct wiphy *wiphy, struct wireless_dev *wdev,
 					int idx, u8 *mac, struct station_info *sinfo)
 {
@@ -5677,3 +5677,3 @@
-	_adapter *padapter = (_adapter *)rtw_netdev_priv(ndev);
+	_adapter *padapter = (_adapter *)rtw_netdev_priv(wdev->netdev);
 	struct sta_info *psta = NULL;
 	struct sta_priv *pstapriv = &padapter->stapriv;
